spec.bs

```diff
@@ -386,49 +396,91 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
 <div algorithm>
   To <dfn>canonicalize protocol</dfn> given a string |value| and a string |type|:
 
-  1. TODO |value| |type|
+  1. Let |strippedValue| be the given |value| with a single trailing U+003A (:) removed, if any.
+  1. If |type| is "pattern" then return |strippedValue|.
+  1. Let |dummyURL| be a new [=url=] record.
+  1. Let |parseResult| be the result of [=Basic URL parser|URL parsing=] |strippedValue|, followed by U+003A (:), with |dummyURL| as [=basic URL parser/url=].
```

Comment:
```suggestion
  1. Let |parseResult| be the result of running the [=basic URL parser=] given |strippedValue| followed by U+003A (:), with |dummyURL| as [=basic URL parser/url=].
```

this was pretty confusing as I thought you were intentionally going through the full parser.

Reply:
Here is a good place to insert a note about why using state override doesn't work.